falcon/util/misc.py

```diff
@@ -79,17 +80,21 @@ def decorator(func):
 if (
     # NOTE(kgriffs): https://bugs.python.org/issue28969
     (sys.version_info.minor == 5 and sys.version_info.micro < 4) or
-    (sys.version_info.minor == 6 and sys.version_info.micro < 1) or
-
-    # PERF(kgriffs): Using lru_cache is slower on pypy when the wrapped
-    #   function is just doing a few non-IO operations.
-    (sys.implementation.name == 'pypy')
+    (sys.version_info.minor == 6 and sys.version_info.micro < 1)
```

Comment:
This would benefit from Python 4 awareness too.

Reply:
Do you mean 3.4? If so, I didn't put this in because we only support 3.5+ for the framework as a whole. Or perhaps you are trolling me about Python 4.0 😛

Reply:
Yes, I did mean 4.0 🙂 (I was semi-serious, why not compare the whole tuple.)

Reply:
Heh, OK. I guess it couldn't hurt.